### PR TITLE
cleanup legal-hold/retention and update docs

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: cmd/client-s3.go [hunk 1]
```diff
@@ -2120,57 +2131,61 @@ func (c *S3Client) ShareUpload(isRecursive bool, expires time.Duration, contentT
...
-	if err != nil {
-		return nil, nil, nil, probe.NewError(err)
+	mode, validity, unit, e := c.api.GetBucketObjectLockConfig(bucket)
+	if e != nil {
+		return "", 0, "", probe.NewError(e).Trace(c.GetURL().String())
 	}
 
-	return mode, validity, unit, nil
+	// FIXME: this is too ugly, fix minio-go
+	vuint64 := uint64(*validity)
```
Comment (kannappanr) [inline cmd/client-s3.go:2187]: Should be something like 

```
var vuint64 uint64
	if validity != nil {
		vuint64 = uint64(*validity)
	}
	var retentionMode minio.RetentionMode
	if mode != nil {
		retentionMode = *mode
	}

	var validityUnit minio.ValidityUnit
	if unit != nil {
		validityUnit = *unit
	}
	return retentionMode, vuint64, validityUnit, nil
```
As `minio-go` returns `nil` when `objectlockconfig` is not set

File: cmd/client-s3.go [hunk 2]
```diff
@@ -2120,57 +2131,61 @@ func (c *S3Client) ShareUpload(isRecursive bool, expires time.Duration, contentT
...
 // SetObjectLockConfig - Set object lock configurataion of bucket.
-func (c *S3Client) SetObjectLockConfig(mode *minio.RetentionMode, validity *uint, unit *minio.ValidityUnit) *probe.Error {
+func (c *S3Client) SetObjectLockConfig(mode minio.RetentionMode, validity uint64, unit minio.ValidityUnit) *probe.Error {
 	bucket, _ := c.url2BucketAndObject()
 
-	err := c.api.SetBucketObjectLockConfig(bucket, mode, validity, unit)
-	if err != nil {
-		return probe.NewError(err)
+	// FIXME: This is too ugly, fix minio-go
+	vuint := (uint)(validity)
```
Comment (kannappanr) [inline cmd/client-s3.go:2138]:
```
var vuint *uint
	if validity != 0 {
		vuint = nil
	}
	var modePtr *minio.RetentionMode
	if mode.IsValid() {
		modePtr = &mode
	}
	var unitPtr *minio.ValidityUnit
	if unit == minio.Days || unit == minio.Years {
		unitPtr = &unit
	}
	e := c.api.SetBucketObjectLockConfig(bucket, modePtr, vuint, unitPtr)

```
Should be something like this. Minio-go rejects if the value is not valid

File: cmd/client-s3.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/client-s3.go]: fixed

File: cmd/client-s3.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/client-s3.go:2138]: fixed

**Comment by kannappanr** [comment]
```
$ mc lock s3/kannappan336
mc: <ERROR> Cannot get object lock configuration on the specified bucket. Invalid arguments provided, please refer `mc <command> -h` for relevant documentation.
$ mc lock s3/kannappan336 -c
mc: <ERROR> Cannot enable object lock configuration on the specified bucket. Invalid arguments provided, please refer `mc <command> -h` for relevant documentation.
```

**Comment by harshavardhana** [comment]
> ```
> $ mc lock s3/kannappan336
> mc: <ERROR> Cannot get object lock configuration on the specified bucket. Invalid arguments provided, please refer `mc <command> -h` for relevant documentation.
> $ mc lock s3/kannappan336 -c
> mc: <ERROR> Cannot enable object lock configuration on the specified bucket. Invalid arguments provided, please refer `mc <command> -h` for relevant documentation.
> ```

Isn't setting empty not allowed @kannappanr ??

**Comment by harshavardhana** [comment]
> ```
> $ mc lock s3/kannappan336
> mc: <ERROR> Cannot get object lock configuration on the specified bucket. Invalid arguments provided, please refer `mc <command> -h` for relevant documentation.
> $ mc lock s3/kannappan336 -c
> mc: <ERROR> Cannot enable object lock configuration on the specified bucket. Invalid arguments provided, please refer `mc <command> -h` for relevant documentation.
> ```

fixed @kannappanr

**Comment by harshavardhana** [comment]
PR is rebased @kannappanr @nitisht

**Pull request merged**